src/LinAlg/hiopVectorRajaImpl.hpp

```diff
@@ -130,28 +132,27 @@ hiopVectorRajaPar::hiopVectorRajaPar(
   n_local_ = glob_iu_ - glob_il_;
 
 #ifndef HIOP_USE_GPU
-  mem_space_ = "HOST"; // If no GPU support, fall back to host!
+  assert(mem_space_ == "HOST"); 
 #endif
 
-  auto& resmgr = umpire::ResourceManager::getInstance();
-  umpire::Allocator devalloc  = resmgr.getAllocator(mem_space_);
-  data_dev_ = static_cast<double*>(devalloc.allocate(n_local_*sizeof(double)));
-  if(mem_space_ == "DEVICE")
+  data_dev_ = exec_space_.template alloc_array<double>(n_local_);
```

Comment:
I'm getting very thrown off by this syntax, and don't know what to google to learn more. Is this an artifact of the templatized execution space class that we are using now?

Reply:
yes, compiler would not confuse the brackets for operators

Reply:
Google for "dot-template": https://stackoverflow.com/questions/8463368/template-dot-template-construction-usage.

I have to agree with @CameronRutherford, this syntax looks awful. :)